### ISSUE TITLE
Feature request  : texture compression support for external texture packs 

### ISSUE THREAD

**Issue by Askmewho** [sm64pc/sm64pc#289]
Nowdays any platform have some kind of texture compression support on the hardware, being s3tc a bit old but the standard (it's available on hardware level on rpi4) it would be cool to have that kind of support to reduce vram usage when using external textures.

**Comment by parkerlreed** [comment]
Wait, external textures is a thing?

**Comment by fgsfdsfgs** [comment]
Yes. Build nightly with `EXTERNAL_DATA=1`.
As for compression, yeah, could make S3TC/other compression types built into GL an option.

**Comment by Askmewho** [comment]
@fgsfdsfgs that would be really helpful for the pi4! Now with s3tc, everything than use it got a boost on performance since mesa add it.

**Comment by parkerlreed** [comment]
Is there any information on the format for this? I would love to try and upscale the base rextures.

**Comment by Askmewho** [comment]
@parkerlreed it's experimental right now I guess.. I would recommend to wait. 

@fgsfdsfgs not only it will benefit RPI 4, many texture pack maker will love the idea of s3tc support (I am in touch with a major one). Thanks buddy to consider it.

**Comment by fgsfdsfgs** [comment]
@parkerlreed for what? External textures? They are stored as plain PNGs.

**Comment by parkerlreed** [comment]
I get that I just meant like file names and stuff. Is it a single image or a bunch of splits or what.

**Comment by fgsfdsfgs** [comment]
There's a bunch of people in one of the Discord servers working on several texture packs, they know the details.
Filenames are the same as they are for unpacked ROM textures, skies are split into 32x32 tiles, everything else is not.

**Comment by parkerlreed** [comment]
Ahh thanks. Very cool. Private discord?

**Comment by coltongit** [comment]
> 
> 
> Ahh thanks. Very cool. Private discord?

nah. Public. Invite's here if you want to join in: https://discord.gg/ZezK78p
